composer.json

```diff
@@ -9,7 +9,7 @@
   },
   "require": {
     "filp/whoops": "^1.1 || ^2.0",
-    "zendframework/zend-diactoros": "^1.1.0 || ^2.0"
+    "middlewares/utils": "^3.0"
```

Comment:
This implicitly raises the (already implicit) minimu PHP version to 7.2, which I would like to avoid if possible. What alternatives are there for `middlewares/utils`?

Could I leave the discovery part completely to clients and simply require injecting a factory to the `WhoopsRunner` class?

(Please rebase the branch on top of the latest master, the build should start failing.)

Reply:
I think the point of the Factory is to be setup and global for a project. I think the utils 3.0 is required because of the mocks tbh so maybe by switching to diactoros in `require-dev` might offer the `utils ^2.0 || ^3.0` version range.

Reply:
Another option is to explicitly statically set a `ResponseFactoryInterface` and `StreamFactoryInterface` from https://github.com/php-fig/http-factory to build a response. The only complete factory interface i know is from middlewares/utils but only from v3 on, which reintroduces that dependency.

Reply:
I want to sleep over this one. Thanks for explaining!